### PR TITLE
Add TCX to codoon_sync.py

### PR DESCRIPTION
**Pull request by lbp0**
add argument --with-tcx

### CODE REVIEW

**Comment by yihong0618** [comment]
Than is cool  thanks, @ben-29 can you help the review?

File: scripts/codoon_sync.py
Comment (yihong0618) [inline scripts/codoon_sync.py:21]: seems we need to add numpy to requirements

File: scripts/codoon_sync.py
Comment (yihong0618) [inline scripts/codoon_sync.py:95]: fit_array can be None here?

**Review by yihong0618 — COMMENTED** [review]
Nice work. we will take test it later

**Comment by yihong0618** [comment]
please fix the lint ~

File: scripts/codoon_sync.py
Comment (lbp0) [reply scripts/codoon_sync.py:95]: yep, in line 191, set None as default

File: scripts/codoon_sync.py
Comment (ben-29) [inline scripts/codoon_sync.py:87]: do we need this function anymore?

File: scripts/codoon_sync.py
Comment (ben-29) [inline scripts/codoon_sync.py:139]: you can use `adjust_time_to_utc` function

File: scripts/codoon_sync.py
Comment (ben-29) [inline scripts/codoon_sync.py:186]: use `GPX_FOLDER` instead

**Review by ben-29 — CHANGES_REQUESTED** [review]
Good job. please check the comments and fix the lint with `black`, Great thanks.

File: scripts/codoon_sync.py
Comment (lbp0) [reply scripts/codoon_sync.py:87]: ok i will remove this

File: scripts/codoon_sync.py
Comment (lbp0) [reply scripts/codoon_sync.py:186]: do i need to export tcx files to gpx folder? and where is that constant please, i didn't find it

Comment (ben-29) [reply scripts/codoon_sync.py:186]: https://github.com/yihong0618/running_page/blob/3ae13b477ef75034ff455ae1fe3163f041afb610/scripts/config.py#L11
Sorry for my typo, it's `TCX_FOLDER`

Comment (lbp0) [reply scripts/codoon_sync.py:186]: got it thx

File: requirements.txt
Comment (yihong0618) [inline requirements.txt:23]: we not not need this, you can use other function in this repo, that can get local time

File: scripts/utils.py
Comment (yihong0618) [inline scripts/utils.py:43]: maybe print something?

**Review by yihong0618 — APPROVED** [review]
LGTM

**Comment by yihong0618** [comment]
@ben-29 most LGTM
@lbp0 Thank you very much, and welcome to open source world~

**Review by ben-29 — COMMENTED** [review]
I have done a little bit of testing, but the tcx files fail to save. 
After changing a few lines of code. It did save tcx files successfully. Well done and do the final change~
You can use my auth-token for testing: 2ec719f753af36a413e2158f82d7d742 fef1a5e9-daac-4af4-bac5-b78d887a4a08

File: scripts/codoon_sync.py
Comment (lbp0) [reply scripts/codoon_sync.py:244]: sorry i didn't notice this lol

**Comment by lbp0** [comment]
> @ben-29 most LGTM
> @lbp0 Thank you very much, and welcome to open source world~

thanks so much,both of you

File: scripts/codoon_sync.py
Comment (lbp0) [reply scripts/codoon_sync.py:231]: I don't decide to add elevations for this time, just step by step. I will do it after finish this merging :P

File: requirements.txt
Comment (lbp0) [reply requirements.txt:23]: no i use it to get ueser's timezone. or we have a function to get this in this repo?

File: scripts/codoon_sync.py
Comment (lbp0) [reply scripts/codoon_sync.py:106]: it works good in my environment so i didn't learn more about this. thank you so much for your advice

File: scripts/codoon_sync.py
Comment (ben-29) [reply scripts/codoon_sync.py:219]: Try run with `--with-tcx --from-auth-token  2ec719f753af36a413e2158f82d7d742 fef1a5e9-daac-4af4-bac5-b78d887a4a08 `, you will see what's wrong.
Do the final kick and we will merge when the tcx saves as expected.

**Comment by lbp0** [comment]
> I have done a little bit of testing, but the tcx files fail to save. After changing a few lines of code. It did save tcx files successfully. Well done and do the final change~ You can use my auth-token for testing: 2ec719f753af36a413e2158f82d7d742 fef1a5e9-daac-4af4-bac5-b78d887a4a08

now all of your runs can be exported successfully. thanks for your token!!!

**Review by ben-29 — APPROVED** [review]
Well done. Thanks for contributing.

**Comment by yihong0618** [comment]
@lbp0 Thanks a lot, its amazing work, we will use your work that add `--with-tcx` to other `sync` either.~

**Pull request merged**
